odoo/orm/environments.py

```diff
@@ -49,12 +50,11 @@ class Environment(Mapping):
     uid: int
     context: frozendict
     su: bool
-    registry: Registry
-    cache: Cache
     transaction: Transaction
 
     def reset(self):
         """ Reset the transaction, see :meth:`Transaction.reset`. """
+        warnings.warn("Since 19.0, use directly `transaction.reset()`", DeprecationWarning)
```

Comment:
I forgot to submit my review before r+. But it is not a big problem
I feel a little bit strange to use the new api `env.transaction.reset()`. I feel the `transaction.reset()` is smentically closer to `transaction.rollback()` instead of `transaction.reset_config/state/data/binding()` because the subject is `transaction` 🤔  while `env.reset()` didn't have this problem

Reply:
`env.reset()` was resetting the transaction (nothing more). Note that is does not rollback anything: it just reconnects to a new registry while keeping the cursor open.

Reply:
The name `reset()` was already present on the transaction, I did not change that. It is a quite technical function. Feel free to rename it in the future.

Reply:
@HydrionBurst `transaction.reset()` should be understood as "fix up the environments in `transaction`" after some operation that made their registry and cache invalid.  I also agree that "reset" may be a bit confusing.